frontend/packages/shared/src/queue/Queue.ts

```diff
@@ -0,0 +1,38 @@
+type QueueTask = {
+  id: string;
+  callback: Function;
+};
+
+type QueueOptions = {
+  timeout?: number;
+};
+
+export class Queue {
+  private queue: QueueTask[] = [];
+  private queueTimeoutId: NodeJS.Timeout | undefined;
+  private readonly options: QueueOptions | undefined;
+
+  constructor(options: QueueOptions) {
+    this.options = options;
+  }
+
+  public addTaskToQueue(queueTask: QueueTask): void {
+    const alreadyInQueue = this.queue.find((task) => task.id === queueTask.id);
+    if (alreadyInQueue) return;
+
+    this.queue.push(queueTask);
+    this.resetTimeOutQueue();
+    this.startProcessingQueue();
+  }
+
+  private resetTimeOutQueue() {
+    clearTimeout(this.queueTimeoutId);
+  }
+
+  private startProcessingQueue() {
```

Comment:
Neat-pick, but should we add return value on this methods?

```suggestion
  private resetTimeOutQueue(): void {
    clearTimeout(this.queueTimeoutId);
  }

  private startProcessingQueue(): void {
```